### PR TITLE
Updated to AGP 8.2.2

### PR DESCRIPTION
**Pull request by LachlanMcKee**
**Description**:
- Updated AGP to 8.2.2

### CODE REVIEW

File: gradle.properties [hunk 1]
```diff
@@ -10,7 +10,7 @@
 # When configured, Gradle will run in incubating parallel mode.
 # This option should only be used with decoupled projects. More details, visit
 # http://www.gradle.org/docs/current/userguide/multi_project_builds.html#sec:decoupled_projects
-org.gradle.jvmargs=-Xmx4g -XX:MaxPermSize=512M
+org.gradle.jvmargs=-Xmx2048m -Dfile.encoding=UTF-8
```
Comment (LachlanMcKee) [inline gradle.properties:13]: Copied from Appyx after Github Actions decided it doesn't like MaxPermSize

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -88,60 +85,70 @@ jobs:
...
-        timeout-minutes: 60
-        env:
-            JAVA_TOOL_OPTIONS: -Xmx4g
-            GRADLE_OPTS: -Dorg.gradle.daemon=false -Dorg.gradle.workers.max=2 -Dkotlin.incremental=false -Dkotlin.compiler.execution.strategy=in-process
-        strategy:
-            fail-fast: false
-            matrix:
-                api-level: [22,25,29]
-        name: Instrumentation tests [API=${{ matrix.api-level }}]
+        name: Instrumentation tests
```
Comment (LachlanMcKee) [inline .github/workflows/ci.yml:88]: Copied from Appyx configuration as the previous RIBs setup was not working (and is very old)

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -9,16 +9,16 @@ jobs:
...
+            - uses: gradle/wrapper-validation-action@v2
+            - uses: actions/setup-java@v4
               with:
-                  java-version: 11
-            - uses: actions/cache@v2
+                distribution: 'zulu'
+                java-version: '17'
+            - name: Setup Gradle
+              uses: gradle/actions/setup-gradle@v3
+            - uses: actions/cache@v4
```
Comment (CherryPerry) [inline .github/workflows/ci.yml:21]: Remove `cache` step for Gradle, it is already set in `gradle/actions/setup-gradle`

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -38,34 +38,37 @@ jobs:
...
```
Comment (CherryPerry) [inline .github/workflows/ci.yml:71]: Remove

File: .github/workflows/ci.yml [hunk 5]
```diff
@@ -88,60 +91,73 @@ jobs:
...
+        - uses: gradle/wrapper-validation-action@v2
+        - uses: actions/setup-java@v4
           with:
-            java-version: 11
-        - uses: actions/cache@v2
+            distribution: 'zulu'
+            java-version: '17'
+        - name: Setup Gradle
+          uses: gradle/actions/setup-gradle@v3
+        - uses: actions/cache@v4
```
Comment (CherryPerry) [inline .github/workflows/ci.yml:160]: Remove

File: gradle/gradle-mvn-push.gradle [hunk 6]
```diff
@@ -11,22 +8,58 @@ def getSnapshotRepositoryUrl() {
...
+                        license {
+                            name.set("The Apache Software License, Version 2.0")
+                            url.set("https://www.apache.org/licenses/LICENSE-2.0.txt")
+                        }
+                    }
+                    developers {
+                        developer {
+                            id.set("badoo")
+                            name.set("Badoo Trading Limited")
+                            email.set("mvicore@team.bumble.com")
```
Comment (CherryPerry) [inline gradle/gradle-mvn-push.gradle:52]: Remove email? Looks like we did not have it before.

**Review by CherryPerry — APPROVED** [review]

File: .github/workflows/ci.yml (hunk 3 above)
Comment (LachlanMcKee) [reply .github/workflows/ci.yml:21]: Good catch. will do!

File: gradle/gradle-mvn-push.gradle (hunk 6 above)
Comment (LachlanMcKee) [reply gradle/gradle-mvn-push.gradle:52]: weird. I swear I copied it from somewhere. will remove

**Pull request merged**